File: dotnet/System/workspace/csharp/adapters/allors.workspace.adapters/workspace/WorkspaceResult.cs
// <copyright file="Workspace.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Adapters
{
    using System.Collections.Generic;

    public class WorkspaceResult : IWorkspaceResult
    {
        public bool HasErrors { get; }

        public IEnumerable<IObject> VersionErrors { get; }

        public IEnumerable<IObject> AccessErrors { get; }

        public IEnumerable<IObject> MissingErrors { get; }
    }
}

File: typescript/modules/libs/system/workspace/adapters/src/lib/workspace/workspace-result.ts
import { IObject, IWorkspaceResult } from '@allors/system/workspace/domain';
export class WorkspaceResult implements IWorkspaceResult {
  get hasErrors(): boolean {
    return (
      this.versionErrors?.length > 0 ||
      this.missingErrors?.length > 0 ||
      this.mergeErrors?.length > 0
    );
  }

  versionErrors: IObject[];

  missingErrors: IObject[];

  mergeErrors: IObject[];

  addMergeError(object: IObject) {
    this.mergeErrors ??= [];
    this.mergeErrors.push(object);
  }
}
